cmd/webhook.go

```diff
@@ -40,110 +39,6 @@ func (e webhookError) Error() string {
-`,
-		Example: `	# Create webhook for CRD of group crew, version v1 and kind FirstMate.
-	# Set type to be mutating and operations to be create and update.
-	kubebuilder alpha webhook --group crew --version v1 --kind FirstMate --type=mutating --operations=create,update
-`,
-		Run: func(_ *cobra.Command, _ []string) {
-			if err := run(options); err != nil {
-				log.Fatal(webhookError{err})
-			}
-		},
-	}
-
-	options.bindFlags(cmd)
-
-	return cmd
-}
-
-var _ commandOptions = &webhookV1Options{}
-
-type webhookV1Options struct {
-	resource    *resource.Options
-	server      string
-	webhookType string
-	operations  []string
-	doMake      bool
-}
-
-func (o *webhookV1Options) bindFlags(cmd *cobra.Command) {
-	cmd.Flags().StringVar(&o.server, "server", "default", "name of the server")
-	cmd.Flags().StringVar(&o.webhookType, "type", "", "webhook type, e.g. mutating or validating")
-	cmd.Flags().StringSliceVar(&o.operations, "operations", []string{"create"},
-		"the operations that the webhook will intercept, e.g. create, update, delete and connect")
-
-	cmd.Flags().BoolVar(&o.doMake, "make", true, "if true, run make after generating files")
-
-	o.resource = &resource.Options{}
-	cmd.Flags().StringVar(&o.resource.Group, "group", "", "resource Group")
-	cmd.Flags().StringVar(&o.resource.Version, "version", "", "resource Version")
-	cmd.Flags().StringVar(&o.resource.Kind, "kind", "", "resource Kind")
-	cmd.Flags().StringVar(&o.resource.Plural, "resource", "", "resource Resource")
-}
-
-func (o *webhookV1Options) loadConfig() (*config.Config, error) {
-	projectConfig, err := config.Load()
-	if os.IsNotExist(err) {
-		return nil, errors.New("unable to find configuration file, project must be initialized")
-	}
-
-	return projectConfig, err
-}
-
-func (o *webhookV1Options) validate(c *config.Config) error {
-	if !c.IsV1() {
-		return fmt.Errorf("webhook scaffolding is no longer alpha for version %s", c.Version)
-	}
-
-	if err := o.resource.Validate(); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func (o *webhookV1Options) scaffolder(c *config.Config) (scaffold.Scaffolder, error) {
-	// Load the boilerplate
-	bp, err := ioutil.ReadFile(filepath.Join("hack", "boilerplate.go.txt")) // nolint:gosec
-	if err != nil {
-		return nil, fmt.Errorf("unable to load boilerplate: %v", err)
-	}
-
-	// Create the actual resource from the resource options
-	var res *resource.Resource
-	switch {
-	case c.IsV1():
-		res = o.resource.NewV1Resource(&c.Config, false)
-	case c.IsV2():
-		res = o.resource.NewResource(&c.Config, false)
-	default:
-		return nil, fmt.Errorf("unknown project version %v", c.Version)
-	}
-
-	return scaffold.NewV1WebhookScaffolder(&c.Config, string(bp), res, o.server, o.webhookType, o.operations), nil
-}
-
-func (o *webhookV1Options) postScaffold(_ *config.Config) error {
-	if o.doMake {
-		err := internal.RunCmd("Running make", "make")
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
 func newWebhookV2Cmd() *cobra.Command {
```

Comment:
Probably you should remove v2 from names for this function and the rest of the functions of this file.

Reply:
IHMO not now at least. Note that we still checking if is V1 to say that is deprecated and we do not know if we will still follow the same V1, V2 approach to creating the V3 in the future. However, I agree that a second phase we could do that.

Reply:
Can we close this one as solved?